server/github.go

```diff
@@ -263,3 +265,44 @@ func (s *Server) areChecksSuccessfulForPr(pr *model.PullRequest, org string) (bo
 	}
 	return false, nil
 }
+
+func (s *Server) createRepoStatus(pr *model.PullRequest, status *github.RepoStatus) {
+	client := NewGithubClient(s.Config.GithubAccessToken)
+	_, _, err := client.Repositories.CreateStatus(context.Background(), pr.RepoOwner, pr.RepoName, pr.Sha, status)
```

Comment:
This creates a new context again, whereas the parent context can be trickled down from `updateBuildStatus`.

Reply:
Good to know. This is all over mattermod honestly.